opus/application/static_media/js/hash.js

```diff
@@ -101,29 +98,39 @@ var o_hash = {
         return hash;
     },
 
-    // part is part of the hash, selections or prefs
-    getSelectionsFromHash: function() {
+    // get both selections and extras (qtype) from hash.
+    getSelectionsExtrasFromHash: function() {
         let hash = o_hash.getHash();
         if (!hash) {
-            return;
+            return [undefined, undefined];
         }
 
         hash = (hash.search('&') > -1 ? hash.split('&') : [hash]);
         let selections = {};  // the new set of pairs that will not include the result_table specific session vars
+        let extras = {}; // store qtype from url
+
         $.each(hash, function(index, pair) {
             let slug = pair.split('=')[0];
             let value = pair.split('=')[1];
 
             if (!(slug in opus.prefs) && value) {
-                if (slug in selections) {
-                    selections[slug].push(value);
+
+                if (slug.match(/qtype/)) {
```

Comment:
We really don't need to use "match" here. match matches anywhere in the string. We want to know if the string starts with "qtype-". Thus I think we should be using `slug.startsWith("qtype-")`.